internal/distributed/utils/util.go

```diff
@@ -30,3 +35,67 @@ func GracefulStopGRPCServer(s *grpc.Server) {
 		<-ch
 	}
 }
+
+func getTLSCreds(certFile string, keyFile string, nodeType string) credentials.TransportCredentials {
+	log.Info("TLS Server PEM Path", zap.String("path", certFile))
+	log.Info("TLS Server Key Path", zap.String("path", keyFile))
+	creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	if err != nil {
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+	}
+	return creds
+}
+
+func EnableInternalTLS(NodeType string) grpc.ServerOption {
+	var Params *paramtable.ComponentParam = paramtable.Get()
+	var serverCfg *paramtable.GrpcServerConfig
+	switch strings.ToLower(NodeType) {
+	case "datacoord":
+		serverCfg = &Params.DataCoordGrpcServerCfg
+	case "datanode":
+		serverCfg = &Params.DataNodeGrpcServerCfg
+	case "indexnode":
+		serverCfg = &Params.IndexNodeGrpcServerCfg
+	case "proxy":
+		serverCfg = &Params.ProxyGrpcServerCfg
+	case "querycoord":
+		serverCfg = &Params.QueryCoordGrpcServerCfg
+	case "querynode":
+		serverCfg = &Params.QueryNodeGrpcServerCfg
+	case "rootcoord":
+		serverCfg = &Params.RootCoordGrpcServerCfg
+	default:
+		log.Error("Unknown NodeType")
+		return grpc.Creds(nil)
+	}
+	certFile := serverCfg.InternalTLSServerPemPath.GetValue()
```

Comment:
can we use one PemPath for every node?
does it necessary to create a path for each node?

Reply:
Modified this.I have moved the InternalTLSconfig from grpcconfig to InternalTLSconfig.This has removed the need for a switch statement here.